widget/richtext.go

```diff
@@ -45,7 +45,10 @@ type RichText struct {
 func NewRichText(segments ...RichTextSegment) *RichText {
 	t := &RichText{Segments: segments}
 	t.Scroll = widget.ScrollNone
+	t.ExtendBaseWidget(t)
+	
 	t.updateRowBounds()
+	t.Refresh()
```

Comment:
This is something to avoid as it will allocate a renderer immediately, it should not be needed for this fix.
Can this be resolved with a smaller scope change?

Reply:
OK, I reverted those changes and proposed a small change inside lineBounds